projects/packages/search/src/class-rest-controller.php

```diff
@@ -86,6 +88,24 @@ public function register_rest_routes() {
 				'permission_callback' => 'is_user_logged_in',
 			)
 		);
+		register_rest_route(
+			'jetpack/v4',
+			'/search/plan/activate',
+			array(
+				'methods'             => WP_REST_Server::EDITABLE,
+				'callback'            => array( $this, 'activate_plan' ),
+				'permission_callback' => 'is_user_logged_in',
+			)
+		);
+		register_rest_route(
+			'jetpack/v4',
+			'/search/plan/deactivate',
+			array(
+				'methods'             => WP_REST_Server::EDITABLE,
+				'callback'            => array( $this, 'deactivate_plan' ),
+				'permission_callback' => 'is_user_logged_in',
+			)
+		);
```

Comment:
Since these routes are being invoked from the WPCOM service, would it make sense to have a more specific permission set than `is_user_logged_in`?

Reply:
Absolutely good catch here. I did check admin privileges before but seen the token is called `JETPACK__ANY_USER_TOKEN`, which is somewhat misnomer. I changed it to just check log in status. Actually, the call is done with admin privileges from WPCOM, and should be checked accordingly.

Reply:
Addressed in 8e1cb973a7